### DISCUSSION TITLE
Is there a configuration for testing with an external device?

### DISCUSSION THREAD

**Discussion by LRV** [barrel/shopify-vite#135] [Q&A]
When developing with `pnpm run dev` it creates files in http://localhost:5173/frontend/entrypoints/ but that isn't accessible to an external device. I can `pnpm run build` but then i have to use shopify cli to upload theme files to the dev theme in order to test on external device. Is there a better process?

**Comment by tylerhellner** [comment]
For live development, you can use [ngrok](https://ngrok.com/docs/agent/) to open a tunnel from your local instance. For most QA, I just use the theme upload process you laid out.

**Reply by knjshimi** [reply]
I might be misinterpreting the problem you're trying to solve, but you could also try vite --watch to test in other devices

**Comment by jonathanmoore** [comment]
@LRV I just added a solution for this...
https://github.com/barrel/shopify-vite/issues/152
